pageserver/src/error.rs

```diff
@@ -0,0 +1,61 @@
+//!
+//! The `error` module provides a general [`Error`] struct representing different types of
+//! encountered errors when running the page server.
+//!
+
+use thiserror::Error;
+
+#[derive(Error)]
+pub enum Error {
```

Comment:
Not sure what are your plans for the usage scope of this var, but still might make sense to name it `PageserverError` at least.

I suspect (but not sure yet), we end up creating multiple `Error` structs/enums in this module: pg errors are a bit different from thread management ones.